src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -79,34 +79,32 @@ export function DiscoverMainRoute(props: Props) {
         const hasUserDataViewValue = await data.dataViews.hasData
           .hasUserDataView()
           .catch(() => false);
-
-        const hasESDataValue =
-          isDev || (await data.dataViews.hasData.hasESData().catch(() => false));
         setHasUserDataView(hasUserDataViewValue);
-        setHasESData(hasESDataValue);
 
-        if (!hasUserDataViewValue || !hasESDataValue) {
-          setShowNoDataPage(true);
-          return;
-        }
+        if (hasUserDataViewValue) {
+          const defaultDataView = await data.dataViews.getDefaultDataView();
 
-        const defaultDataView = await data.dataViews.getDefaultDataView();
+          if (!defaultDataView) {
+            setShowNoDataPage(true);
+            return;
+          }
 
-        if (!defaultDataView) {
-          setShowNoDataPage(true);
-          return;
-        }
+          const { appStateContainer } = getState({ history, uiSettings: config });
+          const { index } = appStateContainer.getState();
+          const ip = await loadIndexPattern(index || '', data.dataViews, config);
 
-        const { appStateContainer } = getState({ history, uiSettings: config });
-        const { index } = appStateContainer.getState();
-        const ip = await loadIndexPattern(index || '', data.dataViews, config);
+          const ipList = ip.list as Array<SavedObject<DataViewAttributes>>;
+          const indexPatternData = resolveIndexPattern(ip, searchSource, toastNotifications);
 
-        const ipList = ip.list as Array<SavedObject<DataViewAttributes>>;
-        const indexPatternData = resolveIndexPattern(ip, searchSource, toastNotifications);
+          setIndexPatternList(ipList);
 
-        setIndexPatternList(ipList);
+          return indexPatternData;
+        }
 
-        return indexPatternData;
+        const hasESDataValue =
+          isDev || (await data.dataViews.hasData.hasESData().catch(() => false));
+        setHasESData(hasESDataValue);
+        setShowNoDataPage(!hasUserDataViewValue || !hasESDataValue);
```

Comment:
Looks like `hasUserDataViewValue` is always `false` here and `hasESDataValue` won't have an effect